### ISSUE TITLE
Items are not being distributed into columns appropriately

### ISSUE THREAD

**Issue by brandonmcconnell** [cedricdelpoux/react-responsive-masonry#117]
I see lots of available space in one column, but extra items are still added to another column

<img src="https://github.com/cedricdelpoux/react-responsive-masonry/assets/5913254/8b438140-acaf-462e-94d5-bab0225de737" width="300" />

Here is my impl:
```tsx
<ResponsiveMasonry
  columnsCountBreakPoints={{ 350: 1, 750: 2, 1100: 3 }}
>
  <Masonry>
    ...
  </Masonry>
</ResponsiveMasonry>
```

**Comment by dpaulos6** [comment]
tested it and it's works fine for me

**Comment by Denis-Andreev** [comment]
Check the source of lib, there are simple logic for elem distribution and working not as you expect

**Comment by walshie4** [comment]
I see this happening as well especially as the dataset grows through an infinite scroll type fetch pattern.  A couple columns end up growing much longer (FWIW our content is different sized which may contribute to the issue)

**Comment by cmastro** [comment]
I have the same issue when the dataset grows in a infinite scroll. If you call to an API and the array of elements grows with differents heights, the items are not being distributed appropriately.

**Comment by cedricdelpoux** [comment]
PR's are welcome if you want to help improving the distribution algorithm

**Comment by cedricdelpoux** [comment]
Published in `2.4.0`
